### PR TITLE
Create ESEntityBuilder to load entities using the repository

### PR DESCRIPTION
**Pull request by LucasMagnum**
Rebuild entity state using the repository.

```python
 >> from eventsourcing_helpers.builders import ESEntityBuilder
 >> builder = ESEntityBuilder(repository_config=config)
 >> builder.rebuild(entity_class=Order, id="123131323132")
        Order(id="123131323132", state="completed", ...)

```

image

### CODE REVIEW

File: eventsourcing_helpers/factories.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
+from typing import Any, Callable, Generator
+
+from eventsourcing_helpers.models import Entity
+from eventsourcing_helpers.repository import Repository
+from eventsourcing_helpers.serializers import from_message_to_dto
+
+
+class ESEntityFactory:
+    """
+        This class is responsible for loading the messages and
```
Comment (simonwahlgren) [inline eventsourcing_helpers/factories.py]: Please de-indent this docstring

File: eventsourcing_helpers/factories.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        self.message_deserializer = message_deserializer
+        self.repository = repository(repository_config)
+
+    def rebuild(self, entity_class: Entity, id: str, max_offset=None) -> Entity:  # noqa
+        """
+        Rebuild the entity state based on the events on the repository.
+
+        Args:
+            entity_class: Class used to apply the events
+            id: entity id used to load the events
```
Comment (simonwahlgren) [inline eventsourcing_helpers/factories.py]: Please add "max_offset" to the Args section

**Comment by LucasMagnum** [comment]
@simonwahlgren yep, I agree that is more of a builder pattern. I will rename it (:

**Comment by simonwahlgren** [comment]
Really good job! For the future we might have to rethink the file structure a bit. I don't like putting the class in a generic `builders.py` file when this is a very specific class. A better structure would maybe be something like this:
```
eventsourcing_helpers/
    models/
        __init__.py
        builder.py
        entity.py
        ...
```
and in the `__init__.py` we could just do:
```py
from eventsourcing_helpers.models.builder import ESEntityBuilder
from eventsourcing_helpers.models.builder import Entity
...
```
But lets discuss this at another time, or when we introduce another builder ;)

**Pull request merged**